packages/helix-shared-storage/src/storage.js

```diff
@@ -524,7 +534,7 @@ class Bucket {
           path: `${key.substring(prefix.length)}`,
         });
       });
-    } while (ContinuationToken);
+    } while (ContinuationToken && objects.length < maxItems);
```

Comment:
first condition stays `false` for `maxItems` POSITIVE_INFINITY, second stays `true`